src/dhcp/aca_dhcp_server.cpp

```diff
@@ -197,4 +211,175 @@ int ACA_Dhcp_Server::_get_db_size() const
   }
 }
 
+void ACA_Dhcp_Server::_init_dhcp_msg_ops()
+{
+  _parse_dhcp_msg_ops[DHCP_MSG_NONE] = _parse_dhcp_none;
+  _parse_dhcp_msg_ops[DHCP_MSG_DHCPDISCOVER] = _parse_dhcp_discover;
+}
+
+void ACA_Dhcp_Server::dhcps_recv(void *message)
+{
+  const dhcp_message *dhcpmsg = nullptr;
```

Comment:
@cj-chung - I am not sure if the dhcp_message format is compatible with your packet_in message from your code, please take a look and try to reach agreement on how the message should look like.

Reply:
The dhcp_message should be the standard udp message format. 
In aca_ovsc_control.cpp, I use following code to identify DHCP packet:
```
        if (udp_sport == 68 && udp_dport == 67) {
            ACA_LOG_INFO("   Message Type: DHCP\n");
        }
```

Once I got a DHCP packet, I can send original packet or whatever message you need to you.